core/src/test/scala/io/finch/EndToEndSpec.scala

```diff
@@ -53,4 +55,15 @@ class EndToEndSpec extends FinchSpec {
     rep.contentString shouldBe "bar"
     rep.status shouldBe Status.Created
   }
+
+  it should "convert Endpoint[AsyncStream[Buf]] to service" in {
```

Comment:
@vkostyukov I finally got it, thank you!
For some reason, I was sure that I can't read anything from ```response.reader``` but I was wrong.

Good news is that new test is failing on ```master``` and passes after this fix.